nimbusapp

```diff
@@ -140,30 +204,68 @@ if [[ -z ${PROJECT} ]]; then
    PROJECT=`echo ${IMAGE}|sed -r 's|\w+/||'|sed 's/:/_/'`
 fi
 
-#Build Command
-if [[ $ACTION == "inspect" ]]; then
-    DOCKERAPP_ACTION="inspect"
-    ISDOCKERAPPACTION=0
-elif [[ $ACTION == "render" ]]; then
-    ISDOCKERAPPACTION=0
-fi
+COMPOSE_ARGS="$@"
+COMPOSE_DIR="${HOME}/.nimbusapp/cache/${PROJECT}/${REPOSITORY}/${IMAGE}/${VERSION}"
+COMPOSE_FILE="${COMPOSE_DIR}/${IMAGE}.yml"
+COMPOSE_BACKUP="${COMPOSE_FILE}.bk"
 
-COMMAND="docker-app ${DOCKERAPP_ACTION}"
+if [[ -n "${DOCKERAPP_ACTION}" ]]; then
+   if curl -s --connect-timeout "${NIMBUS_DOCKERHUB_TIMEOUT}" "${NIMBUS_DOCKERHUB_URL}" > /dev/null 2>&1; then
+       DOCKERAPP_COMMAND="${DOCKERAPP_COMMAND} ${DOCKERAPP_ACTION} ${DOCKERAPP_ARGS} ${REPOSITORY}/${IMAGE}:${VERSION}"
 
-if [[ ! ${ARGS} == "" ]]; then
-    COMMAND="${COMMAND} ${ARGS}"
-fi
+       if [[ "$DOCKERAPP_ACTION" == "render" ]]; then
+          DOCKERAPP_COMMAND="${DOCKERAPP_COMMAND} > ${COMPOSE_FILE}"
+
+          # Backup previous successful attempts in case this try fails
+          # Ignore empty files (assume previous failure)
+          if [[ -s "${COMPOSE_FILE}" ]]; then
+            mv -v "${COMPOSE_FILE}" "${COMPOSE_BACKUP}"
+          else
+            rm -vf "${COMPOSE_FILE}"
+          fi
+       fi # action == render
+
+       mkdir -p "${COMPOSE_DIR}"
+
+       echo
```

Comment:
Is there a specific reason for the extra echo?

Reply:
Spacing of the output, I'll remove that when I look at settings for verbose output